### PR TITLE
feat(cardinal): improve logging

### PR DESCRIPTION
**Pull request by pyrofolium**
Closes: #CAR-149

## What is the purpose of the change
Closed the other PR as there are massive changes. This is a new one. 

This issue injects a logger into system function that already has the system name as a log parameter. 
Additionally added some utility functions to help log additional information about the world, components, systems and entities.

These utility functions do not actually trigger a log. Instead they just augment a zerolog.Event with additional information. The user is suppose to continue to use zerolog methods and patterns to do logging. 

System now has a new type signature that accepts a logger as a parameter. 
There is a new type called RegisteredSystem which is basically the type signature of the old system. 

World curries a new logger into System to form RegisteredSystem then saves it within world. 

## Brief Changelog
Added a new type called RegisteredSystem
relevant files:
     system.go
     world.go 
     world_test.go

Other files were changed simply because changing because the signature of System changed so instantiations of System also needed to be changed the most relevant files with the core changes are listed above.

## Testing and Verifying

Changed all unit tests referencing system to use new function signature
added unit test called TestWorldLogger
test world logger tests:
- a system to see whether it can log correctly and have the system name in the log
- utility functions to print out certain states of various states of certain things.

### CODE REVIEW

**Review by technicallyty — APPROVED** [review]
LGTM

File: cardinal/ecs/world.go
Comment (smsunarto) [inline cardinal/ecs/world.go:119]: lets add a comment here on what this is doing and what it is for

File: cardinal/ecs/world.go
Comment (smsunarto) [inline cardinal/ecs/world.go:784]: can we add comments to all public methods

File: cardinal/ecs/system.go
Comment (smsunarto) [inline cardinal/ecs/system.go:5]: what is registered system for?

Comment (technicallyty) [inline cardinal/ecs/system.go:5]: i don't think we need to export this type

**Comment by pyrofolium** [comment]
@smsunarto this is approved but because of you requested changes I want to get your approval before I merge. If you approve I will merge, otherwise lmk if there are any additional changes you need. Thanks!

**Review by smsunarto — APPROVED** [review]
lgtm, but eventually we should revisit where we are putting world.go and log.go, i think we are really cramming everything in the ecs package right now even things that are not really ecs

**Pull request merged**
